[hip-tests] Update API coverage report generator

fabiomestre opened the pull request:
## Motivation

The aim is to make the HIP API coverage tool more intuitive to use.

## Technical Details

Updates the HIP API coverage tool. It now takes extra arguments for the location of the catch test folder and for the working directory. This avoids issues where the output of the executable is dependent on the path where it is being executed from.

Also updates CmakeLists.txt to integrate seamlessly with the hip-tests project and avoid using commands which rely on relative paths.

## Test Plan

This tool is not covered by CI. Tested locally.

## Test Result

N/A

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

fabiomestre committed:
[hip-tests] Update API coverage report generator

Updates the HIP API coverage tool. It now takes
extra arguments for the location of the catch test folder
and for the working directory. This avoids issues where the output
of the executable is dependent on the path where it is being
executed from.

Also updates CmakeLists.txt to integrate seamlessly with the
hip-tests project and avoid using commands which rely on
relative paths.

rakesroy commented:
There is no testcode change, stress test not needed.

dsicarov-amd commented on projects/hip-tests/utils/coverage/mainCoverage.cpp:
```diff
@@ -23,22 +23,41 @@ THE SOFTWARE.
+    std::cout << "Usage: " << argv[0] << " <hip-include-dir> <catch-test-dir> [work-dir]\n\n"
+              << "Arguments:\n\n"
+              << "  hip-include-dir   Path to the directory containing HIP headers.\n"
+              << "                    Example: <project-root>/projects/hip/include\n\n"
```
Double new line at the the end of string \n\n"

fabiomestre committed:
Remove double new line

fabiomestre committed:
Merge branch 'develop' into fabio/fix_cmake_coverage_tool

fabiomestre committed:
Merge branch 'develop' into fabio/fix_cmake_coverage_tool

AidanBeltonS reviewed (approved):
Change looks good to me. General refactor makes sense

fabiomestre committed:
Merge remote-tracking branch 'origin/develop' into fabio/fix_cmake_coverage_tool

fabiomestre committed:
Remove Cmake option to generate coverage

Removes Cmake option to generate coverage. Instead, explicitly removes
the gen_coverage target from all (this is already the default but
doing it explicitly prevents confusion).

fabiomestre committed:
Merge branch 'develop' into fabio/fix_cmake_coverage_tool

litvaOo commented:
WindowsCI passed in https://mlsejenkinsvm.amd.com/job/rocm-systems/job/hip/view/change-requests/job/PR-1932/6/
